extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/providedchecks/SmallRyeHealthProvidedChecksRecorder.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.smallrye.health.runtime.providedchecks;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class SmallRyeHealthProvidedChecksRecorder {
```

Comment:
I'm not a fan of this pattern (which is indeed how we used to things in Quarkus) and I would much rather see the use of synthetic beans (as it makes things much cleaner and with no chance of race conditions creeping in), but we can do that in a follow up I guess.

Reply:
If I'm right about the fact that the new properties should be runtime, I would be for tackling this issue at the same time too and use synthetic beans right away.

Reply:
Not really sure about properties being runtime, but anyway ok I'll change it to use synthetic beans

Reply:
Or rather one synthetic bean common for all provided health checks that will be produced if at least one provided health check is active. I think it sounds better to have one bean with ~10 properties (where some can simply be unused) than 7 beans with 1-2 properties each

Reply:
Changed it to use a synthetic bean now